lib/phoenix/channel.ex

```diff
@@ -46,43 +46,44 @@ defmodule Phoenix.Channel do
 
   After a client has successfully joined a channel, incoming events from the
   client are routed through the channel's `handle_in/3` callbacks. Within these
-  callbacks, you can perform any action. Typically you'll either forward a
-  message to all listeners with `broadcast!/3`, or push a message directly down
-  the socket with `push/3`. Incoming callbacks must return the `socket` to
-  maintain ephemeral state.
+  callbacks, you can perform any action. Incoming callbacks must return the
+  `socket` to maintain ephemeral state.
 
-  Here's an example of receiving an incoming `"new_msg"` event from one client,
-  and broadcasting the message to all topic subscribers for this socket.
+  Typically you'll either forward a message to all listeners with
+  `broadcast!/3` or reply directly to a client event for request/response style
+  messaging.
+
+  General message payloads are received as maps:
 
       def handle_in("new_msg", %{"uid" => uid, "body" => body}, socket) do
-        broadcast!(socket, "new_msg", %{uid: uid, body: body})
-        {:noreply, socket}
+        ...
+        {:reply, :ok, socket}
       end
 
-  General message payloads are received as maps, and binary data payloads are
-  passed as a `{:binary, data}` tuple:
+  Binary data payloads are passed as a `{:binary, data}` tuple:
 
       def handle_in("file_chunk", {:binary, chunk}, socket) do
         ...
         {:reply, :ok, socket}
       end
 
-  You can also push a message directly down the socket, in the form of a map,
-  or a tagged `{:binary, data}` tuple:
+  ## Broadcasts
+ 
+  Here's an example of receiving an incoming `"new_msg"` event from one client,
+  and broadcasting the message to all topic subscribers for this socket.
 
-      # client asks for their current rank, push sent directly as a new event.
-      def handle_in("current_rank", _, socket) do
-        push(socket, "current_rank", %{val: Game.get_rank(socket.assigns[:user])})
-        push(socket, "photo", {:binary, File.read!(socket.assigns.photo_path)})
+      def handle_in("new_msg", %{"uid" => uid, "body" => body}, socket) do
+        broadcast!(socket, "new_msg", %{uid: uid, body: body})
         {:noreply, socket}
       end
 
   ## Replies
 
-  In addition to pushing messages out when you receive a `handle_in` event,
-  you can also reply directly to a client event for request/response style
-  messaging. This is useful when a client must know the result of an operation
-  or to simply ack messages.
+  Replies are useful for acking a client's message or responding with the
```

Comment:
`acking` may be a difficult term for non-english readers, perhaps we should just use `acknowledge` directly or `respond`?

Could this also use mention of `{:noreply, socket}`? 

> If you do not wish to send a response back to the client, you can return `{:noreply, socket}`

or similar?

Reply:
Changed @rktjmp